### ISSUE TITLE
CurrentLocator observer not called and page progression altered when navigating via hyperlink from the ToC in Epub

### ISSUE THREAD

**Issue by mihai-wolfpack** [readium/kotlin-toolkit#27]
### What happened?

When opening an epub which has a table of contents inside the publication, taping on any of the links takes me to the correct location, however the `currentLocator` observer is not triggered and the progression of the publication is altered, navigating one page back takes me to the first page in the publication, and navigating forward restarts the publication from page 1 after the selected chapter from the ToC has endend. It was observed that on iOS this does not happen and there the behaviour is normal.

### Expected behavior

When taping on a hyperlink of an entry in the ToC, the locator observer should be triggered and the progression / page order of the publication should not be altered in any way.

### How to reproduce?
[Inconjurat_de_idioti.epub.zip](https://github.com/readium/kotlin-toolkit/files/7365648/Inconjurat_de_idioti.epub.zip)

1. Open the publication with an observer set on the `currentLocator` of the navigator 
2. Tap on any of the hyperlinks in the ToC
3. Observe that the locator observer is not called
4. Observe that the page progression is altered when navigating back and forth

### Environment

* Readium version: 2.1.0

#### Development environment
* OS: macOS 11.2
* IDE: Android Studio Arctic Fox 2020.3.1 Patch 2

#### Testing 

* Android version: 10
* Model: OnePlus 5T

* Is it an emulator? No

### Additional context

* Are you willing to fix the problem and contribute a pull request? Yes

**Comment by stevenzeck** [comment]
Yeah that's not great.  The link is simply changing the URL of the current fragment to be that of the resource you selected, which replaces the contents of the fragment/webview.  In this case, the table of contents is completely replaced by whatever chapter you select.

Probably have to intercept the request in [shouldOverrideUrlLoading](https://github.com/readium/kotlin-toolkit/blob/d60d3cbc0fd1bfce2ec24242f1ace75950f054f9/readium/navigator/src/main/java/org/readium/r2/navigator/pager/R2EpubPageFragment.kt#L122) (or [this](https://github.com/readium/kotlin-toolkit/blob/3670f7cf25880e0969c5a418a13b8dd49da74e33/readium/navigator/src/main/java/org/readium/r2/navigator/R2BasicWebView.kt#L478) one) and ask the `navigatorFragment` to go to the resource.

**Comment by mihai-wolfpack** [comment]
I tried looking around to check whether there is any way to intercept the request and use the `go` method instead of it replacing the current URL bu unfortunately i was unable to find a propper solution. Could you please be a little bit more specific or provide such and example?

**Comment by mihai-wolfpack** [comment]
Okay cool, I forked the repository and will open a PR as soon as i manage to fix it. Is this ok?

**Comment by mihai-wolfpack** [comment]
I have created a pull request [here](https://github.com/readium/kotlin-toolkit/pull/28)
